File: truss-compiler-maven-plugin/src/main/java/truss/compiler/generator/bound/xml/BoundClass.java
package truss.compiler.generator.bound.xml;

import javax.xml.bind.annotation.*;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(name = "class")
@XmlAccessorType(XmlAccessType.FIELD)
public class BoundClass {
    @XmlAttribute
    private String name;
    @XmlAttribute
    private String base;
    @XmlAttribute(name = "abstract")
    private boolean abstract_;
    @XmlAttribute
    private boolean ignore;
    @XmlElement(name = "property")
    private List<BoundProperty> properties = new ArrayList<>();
    @XmlElement(name = "superArgument")
    private List<BoundSuperArgument> superArguments = new ArrayList<>();
    private String validation;

    public String getName() {
        return name;
    }

    public String getBase() {
        return base;
    }

    public boolean isAbstract() {
        return abstract_;
    }

    public boolean isIgnore() {
        return ignore;
    }

    public List<BoundProperty> getProperties() {
        return properties;
    }

    public List<BoundSuperArgument> getSuperArguments() {
        return superArguments;
    }

    public String getValidation() {
        return validation;
    }
}

File: Truss.Compiler.Generate/Bound/Xml/BoundClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace Truss.Compiler.Generate.Bound.Xml {
    [XmlRoot("class")]
    public class BoundClass {
        [XmlAttribute("name")]
        public string Name { get; private set; }

        [XmlAttribute("base")]
        public string Base { get; private set; }

        [XmlAttribute("abstract")]
        public bool IsAbstract { get; private set; }

        [XmlAttribute("ignore")]
        public bool IsIgnore { get; private set; }

        [XmlElement("property")]
        public List<BoundProperty> Properties { get; private set; }

        [XmlElement("superArgument")]
        public List<BoundSuperArgument> SuperArgument { get; private set; }

        [XmlElement("validation")]
        public string Validation { get; private set; }
    }
}
